### ISSUE TITLE
Keep previous settings by default?

### ISSUE THREAD

**Issue by AngainorDev** [tloen/alpaca-lora#169]
Thanks for the "Fire" module discover, I'll give it a try, much more straightforward than "Click" I used so far.

https://github.com/tloen/alpaca-lora/blob/88e1130cb552916eab36d0b5e1a19df4685c1e99/finetune.py#L45

I would suggest keeping the previous settings as default:

`train_on_inputs: bool = False`
`group_by_length: bool = False`

I did not check all params, but these two seem important enough to be enabled only if you really want them, not by default.

**Comment by tloen** [comment]
That's reasonable. I'll set group_by_length to default false and cutoff_len to default 256 next time I'm at my computer. Train_on_input=true preserves the old behavior.

**Comment by AngainorDev** [comment]
Just to clarify, because I read several questions around this and I'd like to understand the rational behind it:

Before this commit 
https://github.com/tloen/alpaca-lora/commit/b12c3b90f808e7d62709aad104d4fac1fbc880eb
the prompt was masked in the labels. There was no switch, loss/learning was just on the response part of the sequence.
If I understand it right, the full alpaca training also masks out the input in the labels.
That's the version of the tokenisation I still use for now (with the missing EOS appended to the labels).

After this commit, it became non masked by default.
What is the reason for the change? Is there a quality improvement? Was the masking just not required from the start?
Maybe I missed some relation discussions; could you point me to more details?

**Comment by tloen** [comment]
The prompt wasn't previously masked, actually; my earlier attempts at masking became a no-op because of the DataCollatorForLanguageModeling at the bottom. My mistake here was leaving this non-functional code in the repo for a few days, which misled people as to what it was actually doing.

**Comment by DSXiangLi** [comment]
Hi @tloen ,I would like to confirm that in the original lora finetune, you train on input?

**Comment by xpang-sf** [comment]
May I ask: when run fine-tuning, the setting should be: train_on_inputs: bool = False , right? why train_on_inputs: bool = True  in the code?

**Comment by cfeng16** [comment]
Same question ^^
